hs_core/hydroshare/resource.py

```diff
@@ -747,6 +749,7 @@ def publish_resource(pk):
     resource.raccess.immutable = True
     resource.raccess.save()
     resource.doi = "to be assigned"
+    resource.metadata.create_element('date', type='published', start_date=now().isoformat())
```

Comment:
@pkdash I was thinking about putting metadata modifications into resource_modified() together with other metadata element update like updated date, but you are right, agreed it is better to put this published date create_element() call here rather than in resource_modified(). I will move this statement out from resource_modified() into this publish_resource() then. You can disregard my comment above.